test/facility-guarantee/post-facility-guarantee.api-test.ts

```diff
@@ -0,0 +1,302 @@
+import { PROPERTIES } from '@ukef/constants';
+import { DateStringTransformations } from '@ukef/modules/date/date-string.transformations';
+import { withAcbsAuthenticationApiTests } from '@ukef-test/common-tests/acbs-authentication-api-tests';
+import { IncorrectAuthArg, withClientAuthenticationTests } from '@ukef-test/common-tests/client-authentication-api-tests';
+import { withDateOnlyFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/date-only-field-validation-api-tests';
+import { withNonNegativeNumberFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/non-negative-number-field-validation-api-tests';
+import { withStringFieldValidationApiTests } from '@ukef-test/common-tests/request-field-validation-api-tests/string-field-validation-api-tests';
+import { Api } from '@ukef-test/support/api';
+import { TEST_DATES } from '@ukef-test/support/constants/test-date.constant';
+import { ENVIRONMENT_VARIABLES } from '@ukef-test/support/environment-variables';
+import { RandomValueGenerator } from '@ukef-test/support/generator/random-value-generator';
+import nock from 'nock';
+
+describe('POST /facilities/{facilityIdentifier}/guarantees', () => {
+  const valueGenerator = new RandomValueGenerator();
+  const dateStringTransformations = new DateStringTransformations();
+
+  const facilityIdentifier = valueGenerator.facilityId();
+  const createFacilityGuaranteeUrl = `/api/v1/facilities/${facilityIdentifier}/guarantees`;
+
+  const portfolioIdentifier = PROPERTIES.GLOBAL.portfolioIdentifier;
+  const lenderTypeCode = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.lenderType.lenderTypeCode;
+  const limitTypeCode = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.limitType.limitTypeCode;
+  const sectionIdentifier = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.sectionIdentifier;
+  const guaranteedPercentage = PROPERTIES.FACILITY_GUARANTEE.DEFAULT.guaranteedPercentage;
+
+  const guarantorParty = valueGenerator.stringOfNumericCharacters({ length: 8 });
+  const limitKey = valueGenerator.stringOfNumericCharacters({ length: 8 });
+  const effectiveDateInPast = TEST_DATES.A_PAST_EFFECTIVE_DATE_ONLY;
```

Comment:
Is there any reason to make the names of the constants we're importing so specific? Isn't this just `A_PAST_DATE_ONLY` etc.? This would make them more reusable (see comment below).

Reply:
Left as is, but agree that it could be more universal.